apis/offloading/v1alpha1/namespaceoffloading_types.go

```diff
@@ -0,0 +1,167 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// OffloadingPhaseType represents different namespaces offloading status.
+type OffloadingPhaseType string
+
+const (
+	// ReadyOffloadingPhaseType -> remote Namespaces have been correctly created on previously selected clusters.
+	ReadyOffloadingPhaseType OffloadingPhaseType = "Ready"
+	// NoClusterSelectedOffloadingPhaseType -> no cluster matches user constraints.
+	NoClusterSelectedOffloadingPhaseType OffloadingPhaseType = "NoClusterSelected"
+	// InProgressOffloadingPhaseType -> remote Namespaces' creation is still ongoing.
+	InProgressOffloadingPhaseType OffloadingPhaseType = "InProgress"
+	// OneFailedOffloadingPhaseType -> there was an error during creation of a remote Namespace.
+	OneFailedOffloadingPhaseType OffloadingPhaseType = "OneFailed"
```

Comment:
@palexster Do you think it is worth to keep this or could be merged with `SomeFailed`?